### PR TITLE
Feat/pages and sections

### PR DESCRIPTION
**Pull request by fredrikved**

### CODE REVIEW

File: backend/graph/schema.resolvers.go
Comment (KillerX) [inline backend/graph/schema.resolvers.go:85]: I feel like there is a copy-paste bug here with first and after

File: backend/jsonlogic/filter.go
Comment (KillerX) [inline backend/jsonlogic/filter.go:1]: We should probably add some unit testing here, also around potential SQL injections.

As you note, for now users can't submit this but we should still make sure it's as safe as possible.

Primary reason for unit tests here from my POV is that we don't want to change the results that are produced because that will change the content that is shown. Or at least we should be very aware that we are changing the SQL output here.

File: cms/custom/interface-query-builder/.gitignore
Comment (KillerX) [inline cms/custom/interface-query-builder/.gitignore:2]: This 2 we can probably make sure are in the top level .gitignore.

File: cms/custom/interface-query-builder/src/query-builder/Filter.vue
Comment (KillerX) [inline cms/custom/interface-query-builder/src/query-builder/Filter.vue:1]: Would it make sense to publish this as a npm package somehow?

**Review by KillerX — COMMENTED** [review]
Looking good so far IMO

File: backend/jsonlogic/filter_test.go
Comment (KillerX) [inline backend/jsonlogic/filter_test.go:14]: If you modify the test files a touch and make them something like:

```json
{
    "in": {}, // <- Json logic object
    "out": "X <> 99" // <- Expected result
}
```
then you can just run through all json files in that dir and we can super easily add new cases.

File: backend/jsonlogic/filter.go
Comment (KillerX) [inline backend/jsonlogic/filter.go:37]: Maybe we can use https://pkg.go.dev/github.com/lib/pq#QuoteIdentifier and https://pkg.go.dev/github.com/lib/pq#QuoteLiteral ?

Comment (fredrikved) [reply backend/jsonlogic/filter.go:37]: Oh that's probably better yes

File: cms/custom/interface-query-builder/src/query-builder/Filter.vue
Comment (fredrikved) [reply cms/custom/interface-query-builder/src/query-builder/Filter.vue:1]: We can at least publish the query builder itself as an NPM package when we're happy with it. The directus-specific configuration has to be constructed like this to keep typing, vue capabilities and for directus to correctly register the extension.

For now, it's easier to have it in the same project.

**Review by KillerX — CHANGES_REQUESTED** [review]
There is one comment I would like some clarity on before we merge this but it looks good otherwise IMO.

I didn't do a very deep review of the VUE parts as my VUE experience is gluing together obsolete snippets of code from random places. If you think we need a better review feel free to tag Andreas or maybe even Gerard since he probably has the most experience with Directus.

File: backend/graph/schema.graphqls
Comment (fredrikved) [reply backend/graph/schema.graphqls:47]: At the moment sections are connected to only one page, but could definitely change it for it to be referenced from multiple pages.

Just depends on what a section is, and if it's better for them to just be unique per page. Section in DB is this: 
Title
Description
Type
CollectionId (the source of the content)

Collections can be used by multiple sections.

Comment (fredrikved) [reply backend/graph/schema.graphqls:47]: GQL schema does not 100% represent the DB structure here.

**Review by KillerX — DISMISSED** [review]

**Pull request merged**
